modules/page-memory/src/test/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointerTest.java

```diff
@@ -515,11 +519,112 @@ void testNextCheckpointInterval() throws Exception {
         );
     }
 
+    @Test
+    void testIoSpeedLogsWrittenCorrectly() throws Exception {
+        var metrics = new TrackingCheckpointMetrics();
+
+        Checkpointer checkpointer = createCheckpointerWithPages(metrics);
+
+        CompletableFuture<String> statResult = new CompletableFuture<>();
+
+        logInspector.addHandler(new Handler(
+                event -> event.getMessage().getFormattedMessage().startsWith("Checkpoint finished ["),
+                event -> {
+                    String msg = event.getMessage().getFormattedMessage();
+
+                    String ioSpeed = IO_SPEED_PATTERN.matcher(msg).replaceAll("$1");
+
+                    statResult.complete(ioSpeed);
+                }
+        ));
+
+        assertDoesNotThrow(checkpointer::doCheckpoint);
+
+        await().until(statResult::isDone);
+
+        String ioSpeed = statResult.get(10, SECONDS);
```

Comment:
If there's already an `await` on a previous line then you don't have to pass a timeout anymore, it is meaningless